src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/model/community/GroupedDocuments.kt

```diff
@@ -10,15 +11,20 @@ data class GroupedDocuments(
 }
 
 data class Document(
-  val id: String?,
+  private val _id: String?,
   val documentName: String,
   val author: String?,
   val type: DocumentType,
   val extendedDescription: String?,
   val createdAt: LocalDateTime,
   val lastModifiedAt: LocalDateTime?,
   val parentPrimaryKeyId: Long?,
-)
+) {
+  val id: String
+    get() = this._id ?: UUID.nameUUIDFromBytes(
+      listOf(this.documentName, this.createdAt).joinToString("").toByteArray(),
```

Comment:
as discussed, we should check to see if documentName is immutable to ensure the ID being generated is consistent on subsequent retrieval

Reply:
Do we know why the ID would be null?